staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware.go

```diff
@@ -212,9 +213,15 @@ func (h *UpgradeAwareHandler) ServeHTTP(w http.ResponseWriter, req *http.Request
 		if len(req.URL.RawQuery) > 0 {
 			queryPart = "?" + req.URL.RawQuery
 		}
-		w.Header().Set("Location", req.URL.Path+"/"+queryPart)
-		w.WriteHeader(http.StatusMovedPermanently)
-		return
+		switch method {
+		case http.MethodGet, http.MethodHead:
+			w.Header().Set("Location", req.URL.Path+"/"+queryPart)
+			w.WriteHeader(http.StatusMovedPermanently)
+			return
+		default:
+			path := req.URL.Path
+			req.URL.Path = path + "/" + queryPart
```

Comment:
this is doing a few things that don't seem quite right:

1. it is adding duplicating query data into the path (I think a request with query parameters would get an incorrect path, need to add a unit test to exercise this)
2. it is modifying `req.URL.Path` but not `loc.Path` (which means the `newReq.URL = &loc` code path below is still wrong... need a unit test to exercise that path as well)

My question in https://github.com/kubernetes/kubernetes/pull/95128#discussion_r498275061 was more of a question about behavior ... I'm not actually sure we need to append a `/` ... the go http client might default to sending `/` if there is not path present... if this default case is simply dropped, does the pod still get a `$METHOD / HTTP/1.1` call?